lib/posix/pthread.c

```diff
@@ -302,6 +306,37 @@ int pthread_getschedparam(pthread_t pthread, int *policy,
 	return 0;
 }
 
+/**
+ * @brief Dynamic package initialization
+ *
+ * See IEEE 1003.1
+ */
+int pthread_once(pthread_once_t *once, void (*initFunc)(void))
+{
+	pthread_mutex_lock(&pthread_key_lock);
+
+	if (*once == PTHREAD_ONCE_INIT) {
+		pthread_mutex_unlock(&pthread_key_lock);
+		return 0;
+	} else {
+		(*initFunc)();
+		*once = PTHREAD_ONCE_INIT;
+		pthread_cond_broadcast(&pthread_key_cvar);
+		pthread_mutex_unlock(&pthread_key_lock);
+		return 0;
+	}
+
+	pthread_mutex_lock(&pthread_key_lock);
```

Comment:
All this code from line 328 to 337 is unreachable due to the `else` condition in the previous `if` being a catchall.

Reply:
Oh yeah! Realized I can remove the else block now, will fix this